fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/ReindexService.java

```diff
@@ -113,6 +115,21 @@ public void indexOcflObject(final Transaction tx, final String ocflId) {
                 final var headers = new ResourceHeadersAdapter(storageHeaders);
 
                 final var fedoraId = headers.getId();
+
+                if (config.isRebuildContinue()) {
+                    try {
+                        ocflIndex.getMapping(tx, fedoraId);
+                        // We got the mapping, so we can skip this resource.
+                        throw new ObjectExistsInOcflIndexException(
+                                String.format("Skipping indexing of %s in transaction %s, because" +
+                                " it already exists in the index.", fedoraId, tx.getId())
+                        );
+                    } catch (FedoraOcflMappingNotFoundException e) {
+                        LOGGER.debug("Indexing object {} in transaction {}, because it does not yet exist in the " +
+                                "index.", fedoraId, tx.getId());
+                    }
+                }
+
```

Comment:
Do you know if it's possible to be in a state where a resource has only been partially indexed? I was testing yesterday and I from what it looks like the transaction is tied to the entire object so it _should_ all be committed (or rolled back) at the same time, so I don't think this will be an issue but I haven't seen how larger indexing operations look after they've stopped. I might try to make a stress test later to see if I can find out at all.

Reply:
My understanding is that the entire object is indexed or not indexed in a single transaction. So all of the various indexes should either be updated or not. But I would be happy to have you press on this to ensure I have not underestimated the problem and only partially solved the issue.

Reply:
I haven't been able to produce any partially indexed objects, although my test dataset ended up being a little smaller than I'd liked. Given that this is working for NLM, and that the logic indexes each object in a single transaction I'm ok with this.

An unrelated note, and something I need to make a jira for, is that sufficiently large objects (e.g. archival group with many children/versions) can cause the reindex transaction to timeout. I'm not sure what type of limit for an atomic resource would cause this but I saw it with an archival group.